abstract.go

```diff
@@ -187,6 +195,105 @@ func scrapeDiscoveryJob(job job, tagsOnMetrics exportedTagsOnMetrics, clientTag
 	return awsInfoData, cw
 }
 
+func scrapeDiscoveryJobUsingMetricData(job job, tagsOnMetrics exportedTagsOnMetrics, clientTag tagsInterface, clientCloudwatch cloudwatchInterface) (awsInfoData []*tagsData, cw []*cloudwatchData) {
+	mux := &sync.Mutex{}
+	var wg sync.WaitGroup
+	var getMetricDatas []cloudwatchData
+	var length int
+
+	if job.Length == 0 {
+		length = 120
+	} else {
+		length = job.Length
+	}
+
+	resources, err := clientTag.get(job)
+	if err != nil {
+		log.Println("Couldn't describe resources: ", err.Error())
+		return
+	}
+	commonJobDimensions := getAwsDimensions(job)
+
+	for i := range resources {
+		resource := resources[i]
+		mux.Lock()
+		awsInfoData = append(awsInfoData, resource)
+		mux.Unlock()
+		metricTags := resource.metricTags(tagsOnMetrics)
+		dimensions := detectDimensionsByService(resource.Service, resource.ID, clientCloudwatch)
+		for _, commonJobDimension := range commonJobDimensions {
+			dimensions = append(dimensions, commonJobDimension)
+		}
+
+		wg.Add(len(job.Metrics))
+		go func() {
+			for j := range job.Metrics {
```

Comment:
I think here the semaphore for the Cloudwatch concurrency is missed. Else it can provoque throttling.

It should be: 
```
		go func() {
			cloudwatchSemaphore <- struct{}{}
			defer func() {
				<-cloudwatchSemaphore
			}()
			for j := range job.Metrics {
```